examples/kubectl.rs

```diff
@@ -0,0 +1,169 @@
+//! This is a simple imitation of the basic functionality of kubectl
+//! Supports kubectl {get, delete, watch} <resource> [name] (name optional) with labels and namespace selectors
+use anyhow::{bail, Result};
+use clap::{arg, command};
+use either::Either;
+use futures::{StreamExt, TryStreamExt};
+use k8s_openapi::chrono::{Duration, Utc};
+use kube::{
+    api::{Api, DynamicObject, ListParams, ObjectMeta, Resource, ResourceExt},
+    discovery::{ApiCapabilities, ApiResource, Discovery, Scope},
+    runtime::{
+        utils::try_flatten_applied,
+        wait::{await_condition, conditions::is_deleted},
+        watcher,
+    },
+    Client,
+};
+use log::info;
+use std::{collections::BTreeMap, env};
+
+type KindMap = BTreeMap<String, (ApiResource, ApiCapabilities)>;
+async fn discover_valid_inputs(client: Client) -> Result<KindMap> {
+    let mut kinds = BTreeMap::new(); // valid inputs (keys are plural or kind)
+    let discovery = Discovery::new(client).run().await?;
+    // NB: relies on sort order of discovery groups for precedence (which is not perfect)
+    for group in discovery.groups() {
+        for (ar, caps) in group.recommended_resources() {
+            // two entries per (TODO: extend ApiResource with shortname - its in APIResource..)
+            kinds
+                .entry(ar.kind.to_lowercase())
+                .or_insert((ar.clone(), caps.clone()));
+            kinds.entry(ar.plural.to_lowercase()).or_insert((ar, caps));
+        }
+    }
+    Ok(kinds)
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    // 0. init
+    std::env::set_var("RUST_LOG", "info,kube=info");
+    env_logger::init();
+    let client = Client::try_default().await?;
+
+    // 1. arg parsing
+    let matches = command!()
```

Comment:
Any reason to use the stringly typed mode here rather than the `clap::Parser` macro?

Reply:
I think that brings in the clap_derive crate and was trying to keep the extra dependencies small since it can impacts ci times. Have not really measured though, just know derive stuff usually push it up.

Reply:
I don't think this will have a huge impact since we bring in the proc macro ecosystem for the CI jobs via kube-derive anyway.

Reply:
Actually, looks like `tower` brings it in too...